fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/DataServlet.java

```diff
@@ -346,41 +345,60 @@ public void getData(
     ) {
         try {
             RequestLog.startTiming("DataApiRequest");
-            DataApiRequest apiRequest = new DataApiRequest(
-                    tableName,
-                    timeGrain,
-                    dimensions,
-                    metrics,
-                    intervals,
-                    filters,
-                    havings,
-                    sorts,
-                    count,
-                    topN,
-                    format,
-                    timeZone,
-                    asyncAfter,
-                    perPage,
-                    page,
-                    uriInfo,
-                    this
-            );
+            DataApiRequest apiRequest;
+            try {
+                apiRequest = new DataApiRequest(
+                        tableName,
+                        timeGrain,
+                        dimensions,
+                        metrics,
+                        intervals,
+                        filters,
+                        havings,
+                        sorts,
+                        count,
+                        topN,
+                        format,
+                        timeZone,
+                        asyncAfter,
+                        perPage,
+                        page,
+                        uriInfo,
+                        this
+                );
+            } finally {
+                RequestLog.stopTiming("DataApiRequest");
+            }
 
             if (requestMapper != null) {
                 apiRequest = (DataApiRequest) requestMapper.apply(apiRequest, containerRequestContext);
             }
 
-            RequestLog.switchTiming("DruidQueryMerge");
+            RequestLog.startTiming("DruidQueryMerge");
+            TemplateDruidQuery templateQuery;
             // Build the query template
```

Comment:
You're gonna hate me for this, but in these refactor spots, can we float the comment to the start of the "block" (ie. above the start of the timer)? Having it in the middle is slightly confusing.